driver.c

```diff
@@ -142,8 +133,7 @@ int main(int argc, char **argv)
         break;
     }
     default:
-        printf("Abnormal: unknown request: %d\n", req);
-        return -1;
+        FATAL(-1, "Abnormal: unknown request: %d\n", req);
         break;
```

Comment:
It is interesting because compilers usually provide "shortcuts" to handle unreachable switch-case. Take GCC for example. There is a builtin, `__builtin_unreachable`, which eliminates useless jumps caused by switch statement.